### PR TITLE
Posts#edit/update and minor changes to restaurants/show.html.erb

### PR DESCRIPTION
**Pull request by PracticalEscapement**
Edit and Update actions were added to Posts controller.
Restaurants/show.html.erb needed to have a post ID passed to the form.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -7,10 +7,18 @@ def index
 
   def show
     @post = Post.find(params[:id])
-    @restaurant = Restaurant.find(params[:restaurant_id])
+    restaurant = @post.restaurant_id
+    @restaurant = Restaurant.find(restaurant)
```
Comment (jamesjason) [inline app/controllers/posts_controller.rb]: how about 
```
@restaurant = @post.restaurant
```

this should work if you have `belongs_to :restaurant` in your post model

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -7,10 +7,18 @@ def index
...
   end
 
   def edit
+    @post = Post.find(params[:id])
+  end
+
+  def update
+    @post = Post.find(params[:id])
+    @post.update(title: params[:post][:title], review: params[:post][:review])
```
Comment (jamesjason) [inline app/controllers/posts_controller.rb]: @post.update returns true or false based on if the post was successfully updated or not. I'd add a conditional here to handle the case where update fails

**Review by jamesjason — CHANGES_REQUESTED** [review]
had two small comments but looks good otherwise

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (PracticalEscapement) [reply app/controllers/posts_controller.rb]: I didn't have that association set up, which is why @post.restaurant wasn't accessible. Ill make that correction.

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (PracticalEscapement) [reply app/controllers/posts_controller.rb]: Makes sense, will do.

**Review by jamesjason — APPROVED** [review]
LGTM! 👌

**Pull request merged**